src/main/java/DukeManager.java

```diff
@@ -1,35 +1,145 @@
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.util.Scanner;
+
 public class DukeManager {
     private static final String HORIZONTAL_LINE = "____________________________________________________________\n";
     private static final String ADD_CONFIRMATION_MESSAGE = "Got it. I've added this task:";
-    private static Task[] inputs = new Task[100];
-    private static int index = 0;
+    private static final Path directoryPath = Paths.get(".","data");
+    private static final Path filePath = Paths.get(".","data", "duke.txt");
+    private static ArrayList<Task> inputs = new ArrayList<>();
 
     /**
      * Returns a string with formatted user inputs.
      *
-     * @param inputs list of user input.
      * @return Formatted string to print
      */
-    public static String formatList(Task[] inputs) {
+
+    public static String formatList() {
         String formattedString = HORIZONTAL_LINE;
-        for (int i = 0; i < inputs.length; i++) {
-            if (inputs[i] == null) {
+        for (int i = 0; i < inputs.size(); i++) {
+            if (inputs.get(i) == null) {
                 break;
             }
-            formattedString += (i+1) + "." + inputs[i].toString() + "\n";
+            formattedString += (i+1) + "." + inputs.get(i).toString() + "\n";
         }
         formattedString += HORIZONTAL_LINE;
         return formattedString;
     }
 
+    public static String saveList() {
+        String formattedString = "";
+        for (int i = 0; i < inputs.size(); i++) {
+            if (inputs.get(i) == null) {
+                break;
+            }
+            formattedString += inputs.get(i).toSaveString() + "\n";
+        }
+        return formattedString;
+    }
+
+    public static void checkExists() throws IOException {
+        boolean directoryExists = Files.exists(directoryPath);
+        boolean fileExists = Files.exists(filePath);
+        if (!directoryExists) {
+            Files.createDirectories(directoryPath);
+        }
+        if (!fileExists) {
+            Files.createFile(filePath);
+        }
+    }
+
+    public static void save() {
+        try {
+            checkExists();
+        } catch (IOException e) {
+            print("Something went wrong when trying to save: " + e.getMessage());
+            return;
+        }
+        String pathName = "./data/duke.txt";
```

Comment:
Since you're already using `File` above, there could be some consistency here instead of `String`?

https://nus-cs2113-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#do-not-trip-up-reader